[Matthew Lee] Duke Increments

crazoter opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

crazoter committed:
Level-1

crazoter committed:
Level-2

crazoter committed:
Level-3

crazoter committed:
Level-4

crazoter committed:
Level-4 modifications to shift logic of removing commands out of the class and into Duke (for easier error management later on)

crazoter committed:
A-TextUiTesting, modified abit to output as UTF-8

crazoter committed:
Level-5

crazoter committed:
Level-6

crazoter committed:
Level-7 (but test cases not updated)

crazoter committed:
Level-8

crazoter committed:
Merge branch 'branch-Level-7' into branch-Level-8

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/TaskWithDate.java

crazoter committed:
A-MoreOOP

crazoter committed:
Merge branch 'gradle'

crazoter committed:
Added gradle, checkstyle, shadowjar, junit

crazoter committed:
Added javadocs

crazoter committed:
Implemented find command

crazoter committed:
Merge branch 'branch-Level-9' into branch-A-JavaDoc

crazoter committed:
Patched storage bug

jcjjjared reviewed (commented):
Hi, your code is very clean and the classes are well-organised! Keep up the good work. Generally no syntax violations too! Easy to read and understand.😄 Cheers!

dalisc commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,36 @@
+        for (int i = 0; i < taskList.size(); i++) {
+            String taskDetails = taskList.get(i).toString();
+            if (taskDetails.contains(remainingLine)) {
+                ui.println(i + "." + taskDetails);
```
It wasn't explicitly specified in the instructions whether to assign new indexes to the matching tasks (numbering them from 1 again), but it does imply that in the instructions.

As for me, I kept the original indexes of the task so that the user can more easily refer to their index if they want to issue a Done or Delete command.

The instructions are ambiguous and this is something I will clarify with Prof and my tutor, so just a heads up to keep a lookout for clarifications regarding this!

dalisc commented on src/main/java/duke/command/IndexBasedCommand.java:
```diff
@@ -0,0 +1,17 @@
+package duke.command;
+
+import duke.*;
```
Do avoid using * in imports!

dalisc commented:
Hi! Your code is very clean and readable. I like that you went the extra mile to do the optional increments of making it even more OOP style and packaging them.

I have pointed out some small errors such as the use of asterisks. Nothing serious - just coding standard stuff. Also, commit messages should be in present tense in an imperative style (as taught in Week 3 lecture), so that is something to take note of in the future. Aside from that, you could also write documentation at the class level when time allows. 

All in all, really neat and solid work! Keep it up!

jcjjjared commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,110 @@
+import java.util.Scanner;
+
+public class Ui {
+    private boolean enabled;
```
Could use "isEnabled" instead to make it sound more like a boolean

crazoter committed:
Add GUI except for ExitCommand

Added all 4 GUI tutorials and hooked it up with Duke (with the exception of the exit command as I presume there is a tutorial on it later on)

crazoter committed:
Update shadowjar, update JUnit tests, add unicode.java

Unicode.java to consolidate unicode characters in 1 place.

crazoter committed:
Add assertions

More will be added if necessary

crazoter committed:
Update to fit checkstyle

crazoter committed:
Modify Storage for Stream and Lambda

crazoter committed:
Revamp error messages

crazoter committed:
Add space to JavaDocs description, resolve peer reviews

crazoter committed:
Merge pull request #5 from crazoter/branch-A-CodeQuality

Add space to JavaDocs description, resolve peer reviews

crazoter committed:
Change GUI

Made image circle, fix resize layout issues as well as text clipping

crazoter committed:
Merge pull request #6 from crazoter/branch-ImproveGui

Change GUI

crazoter committed:
Add DialogBox margin

crazoter committed:
Merge pull request #7 from crazoter/branch-ImproveGui

Add DialogBox margin

crazoter committed:
Set theme jekyll-theme-tactile

crazoter committed:
Add autocomplete

crazoter committed:
Merge branch 'master' of https://github.com/crazoter/duke

crazoter committed:
Add user guide

crazoter committed:
Change build.gradle to jar release version

crazoter committed:
Update build.gradle for cross platform jar

crazoter committed:
Revamp Gui

More styling and improve resizeability

crazoter committed:
Cleanup code and build.gradle

crazoter committed:
Update user guide, minor ui improvements

crazoter committed:
Update user guide

crazoter committed:
Update user guide

Update read me

Modifications to make it even more readable and resize image

Update user guide

crazoter committed:
Update read me

Modifications to make it even more readable and resize image

crazoter committed:
Update user guide

crazoter committed:
Merge branch 'master' of https://github.com/crazoter/duke
